src/openforms/js/components/admin/form_design/registrations/objectsapi/utils.js

```diff
@@ -48,9 +48,20 @@ const FORMAT_TYPE_MAP = {
 /**
  * Return a JSON Schema definition matching the provided variable.
  * @param {Object} variable - The current variable
+ * @param {Object} components - The components available in the form. The key is the component key, the value is the
+ * component definition.
  * @returns {Object} - The JSON Schema
  */
-const asJsonSchema = variable => {
+const asJsonSchema = (variable, components) => {
+  // Figure out if the component is a file component (special case)
+  const componentDefinition = components[variable.key];
+  if (componentDefinition && componentDefinition.type === 'file') {
+    // If it is, and it has multiple == True, then type is array
+    if (componentDefinition.multiple) return {type: 'array'};
+    // Otherwise it's string (URL of the document)
+    return {type: 'string', format: 'uri'};
+  }
```

Comment:
I left this as a 'special case'. I thought that if in the future we have more special cases we can refactor this.

Reply:
I'm wondering if this special case should be moved in the `ObjectsApiVariableConfigurationEditor` component instead :thinking:, as it is purely hardcoding a json schema.

Or maybe instead of passing the array of `components`, only pass the relevant `component` to this function?

Reply:
more accurate would also be:

```yaml
type: array
items:
  type: string
  format: uri
```

Reply:
I think leaving it in here is more appropriate, it fits the idea of providing the json schema for a given variable. The fact that the schema is hard coded doesn't really matter, it's the same how your fallback is a hardcoded `type: string`, no?